Bug Expanding Resources

Not had time to properly look into this but will do later and update.

Navigate: Sub -> Databricks -> VNET -> CheckIpAddressAvailability -> CRASH

```
AzBrowse is waiting for access to '~/.azbrowse.db', do you have another instance of azbrowse open?
Loading db complete
2019/11/19 14:56:15 Current version is the latest
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x886fa2]

goroutine 1 [running]:
github.com/lawrencegripper/azbrowse/pkg/armclient.(*Client).DoRequestWithBody(0x0, 0x1518280, 0xc00058c480, 0xec8afd, 0x3, 0xc0013760c0, 0xbd, 0x0, 0x0, 0x0, ...)
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/pkg/armclient/armclient.go:114 +0x422
github.com/lawrencegripper/azbrowse/pkg/armclient.(*Client).DoRequest(...)
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/pkg/armclient/armclient.go:96
github.com/lawrencegripper/azbrowse/internal/pkg/expanders.(*DefaultExpander).Expand(0x1bea278, 0x1518280, 0xc00058c480, 0xc0013ce000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/internal/pkg/expanders/default.go:49 +0xb6
github.com/lawrencegripper/azbrowse/internal/pkg/expanders.ExpandItem(0x1518280, 0xc000454060, 0xc0013ce000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/internal/pkg/expanders/expander.go:113 +0x10c1
github.com/lawrencegripper/azbrowse/internal/pkg/views.(*ListWidget).ExpandCurrentSelection(0xc000d39000)
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/internal/pkg/views/list.go:221 +0x28a
github.com/lawrencegripper/azbrowse/internal/pkg/keybindings.ListExpandHandler.Fn.func1(0xc000b31cb0, 0xc000237770, 0xa, 0xc00110dd01)
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/internal/pkg/keybindings/listhandlers.go:309 +0x2a
github.com/stuartleeks/gocui.(*Gui).execKeybindings(0xc000b31cb0, 0xc000237770, 0xc00110ddb8, 0x700000000, 0x2, 0x3)
        /home/jjcollinge/code/go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:629 +0xc2
github.com/stuartleeks/gocui.(*Gui).onKey(0xc000b31cb0, 0xc00110ddb8, 0x2, 0x0)
        /home/jjcollinge/code/go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:593 +0x19b
github.com/stuartleeks/gocui.(*Gui).handleEvent(...)
        /home/jjcollinge/code/go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:413
github.com/stuartleeks/gocui.(*Gui).MainLoop(0xc000b31cb0, 0xc000454060, 0xc000b31cb0)
        /home/jjcollinge/code/go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:373 +0x1e2
main.main()
        /home/jjcollinge/code/go/src/github.com/lawrencegripper/azbrowse/cmd/azbrowse/main.go:78 +0x231
```

--- a/internal/pkg/expanders/default.go
+++ b/internal/pkg/expanders/default.go
@@ -26,9 +26,6 @@ func (e *DefaultExpander) setClient(c *armclient.Client) {
 	e.client = c
 }
 
-// DefaultExpanderInstance provides an instance of the default expander for use
-var DefaultExpanderInstance DefaultExpander
-
 // Name returns the name of the expander
 func (e *DefaultExpander) Name() string {
 	return "GenericExpander"

--- a/internal/pkg/expanders/expander.go
+++ b/internal/pkg/expanders/expander.go
@@ -108,9 +108,9 @@ func ExpandItem(ctx context.Context, currentItem *TreeNode) (*ExpanderResponse, 
 	}
 
 	// Use the default handler to get the resource JSON for display
-	defaultExpanderWorksOnThisItem, _ := DefaultExpanderInstance.DoesExpand(ctx, currentItem)
+	defaultExpanderWorksOnThisItem, _ := GetDefaultExpander().DoesExpand(ctx, currentItem)
 	if !hasPrimaryResponse && defaultExpanderWorksOnThisItem {
-		result := DefaultExpanderInstance.Expand(ctx, currentItem)
+		result := GetDefaultExpander().Expand(ctx, currentItem)
 		if result.Err != nil {
 			eventing.SendStatusEvent(eventing.StatusEvent{
 				InProgress: true,

--- a/internal/pkg/expanders/registerExpanders.go
+++ b/internal/pkg/expanders/registerExpanders.go
@@ -5,10 +5,16 @@ import (
 )
 
 var swaggerResourceExpander *SwaggerResourceExpander
+var defaultExpander *DefaultExpander
 
 // GetSwaggerResourceExpander returns the (singleton) instance of SwaggerResourceExpander
 func GetSwaggerResourceExpander() *SwaggerResourceExpander {
 	return swaggerResourceExpander
+}
+
+// GetDefaultExpander returns the (singleton) instance of DefaultExpander
+func GetDefaultExpander() *DefaultExpander {
+	return defaultExpander
 }
 
 // register tracks all the current handlers
@@ -21,6 +27,10 @@ var register []Expander
 func InitializeExpanders(client *armclient.Client) {
 	swaggerResourceExpander = NewSwaggerResourcesExpander()
 	swaggerResourceExpander.AddAPISet(NewSwaggerAPISetARMResources(client))
+
+	defaultExpander = &DefaultExpander{
+		client: client,
+	}
 
 	register = []Expander{
 		&TenantExpander{

--- a/pkg/armclient/armclient.go
+++ b/pkg/armclient/armclient.go
@@ -27,7 +27,7 @@ type Client struct {
 	client   *http.Client
 	tenantID string
 
-	aquireToken TokenFunc
+	acquireToken TokenFunc
 }
 
 // LegacyInstance is a singleton ARMClient used while migrating to the
@@ -37,16 +37,16 @@ var LegacyInstance = NewClientFromCLI()
 // NewClientFromCLI creates a new client using the auth details on disk used by the azurecli
 func NewClientFromCLI() *Client {
 	return &Client{
-		aquireToken: aquireTokenFromAzCLI,
-		client:      &http.Client{},
+		acquireToken: aquireTokenFromAzCLI,
+		client:       &http.Client{},
 	}
 }
 
 // NewClientFromClientAndTokenFunc create a client for testing using custom token func and httpclient
 func NewClientFromClientAndTokenFunc(client *http.Client, tokenFunc TokenFunc) *Client {
 	return &Client{
-		aquireToken: tokenFunc,
-		client:      client,
+		acquireToken: tokenFunc,
+		client:       client,
 	}
 }
 
@@ -59,7 +59,7 @@ func (c *Client) SetClient(newClient *http.Client) {
 // SetAquireToken lets you override the token func for testing
 // or other purposes
 func (c *Client) SetAquireToken(aquireFunc func(clearCache bool) (AzCLIToken, error)) {
-	c.aquireToken = aquireFunc
+	c.acquireToken = aquireFunc
 }
 
 // GetTenantID gets the current tenandid from AzCli
@@ -69,7 +69,7 @@ func (c *Client) GetTenantID() string {
 
 // GetToken gets the cached cli token
 func (c *Client) GetToken() (AzCLIToken, error) {
-	return c.aquireToken(false)
+	return c.acquireToken(false)
 }
 
 // RequestResult used with async channel
@@ -111,7 +111,7 @@ func (c *Client) DoRequestWithBody(ctx context.Context, method, path, body strin
 		return "", errors.New("Failed to create request for body: " + err.Error())
 	}
 
-	cliToken, err := c.aquireToken(false)
+	cliToken, err := c.acquireToken(false)
 	if err != nil {
 		return "", errors.New("Failed to acquire auth token: " + err.Error())
 	}
@@ -127,7 +127,7 @@ func (c *Client) DoRequestWithBody(ctx context.Context, method, path, body strin
 	if response != nil && response.StatusCode == 401 {
 		// This might be because the token we've cached has expired.
 		// Get a new token forcing it to clear cache
-		cliToken, err := c.aquireToken(true)
+		cliToken, err := c.acquireToken(true)
 		if err != nil {
 			return "", errors.New("Failed to acquire auth token: " + err.Error())
 		}

